src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/samplers/MetricsDBFileSampler.java

```diff
@@ -0,0 +1,83 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.samplers;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.collectors.SampleAggregator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.emitters.ISampler;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class MetricsDBFileSampler implements ISampler {
+  private static final Logger LOG = LogManager.getLogger(MetricsDBFileSampler.class);
+  private final AppContext appContext;
+
+  public MetricsDBFileSampler(final AppContext appContext) {
+    Objects.requireNonNull(appContext);
+    this.appContext = appContext;
+  }
+
+  @Override
+  public void sample(SampleAggregator sampleCollector) {
+    String metricsdbPrefix = MetricsDB.getFilePrefix();
+
+    Path targetDirectoryPath = Paths.get(metricsdbPrefix).getParent();
+
+    String metricsdbPattern = "regex:" + metricsdbPrefix + "\\d+";
+    String metricsdbTarPattern = "regex:" + metricsdbPrefix.substring(0, metricsdbPrefix.length() - 1)
+        + ".tar.\\d+-\\d+-\\d+-\\d+-\\d+.gz";
+    PathMatcher metricsdbMatcher = FileSystems.getDefault().getPathMatcher(metricsdbPattern);
+    PathMatcher metricsdbTarMatcher = FileSystems.getDefault().getPathMatcher(metricsdbTarPattern);
+    DirectoryStream.Filter<Path> metricsdbFilter = metricsdbMatcher::matches;
+    DirectoryStream.Filter<Path> metricsdbTarFilter = metricsdbTarMatcher::matches;
```

Comment:
Instead of performing a `FileSystems.getDefault().getPathMatcher(metricsdbPattern);`, can use a config to define the location of these files ?  We are performing something similar here : https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/master/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java#L351

Reply:
Not completely sure what you mean by "config" here. There currently a configurable prefix (an absolute path prefix) for these metricsdb files. We probably don't want to split this prefix into a directory and filename prefix if that's what you mean -- doesn't seem too compelling to take on this new addition just for these two uses at the moment.

Reply:
I refer to using a variable like `DB_FILE_PREFIX_PATH_CONF_NAME` used [here](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/master/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java#L351) and my concern was regarding the ` FileSystems.getDefault().getPathMatcher(metricsdbPattern);`. I see that you have addressed this in the latest code revision. 

```
String prefix = PluginSettings.instance().getSettingValue(DB_FILE_PREFIX_PATH_CONF_NAME, DB_FILE_PREFIX_PATH_DEFAULT);
```